connectors/sources/sharepoint.py

```diff
@@ -118,71 +123,102 @@ def __init__(self, configuration):
 
     @classmethod
     def get_default_configuration(cls):
-        """Get the default configuration for Sharepoint
+        """Get the default configuration for SharePoint
 
         Returns:
             dictionary: Default configuration.
         """
         return {
-            "is_cloud": {
-                "value": False,
-                "label": "True if Sharepoint Online, False if Sharepoint Server",
-                "type": "bool",
+            "data_source": {
+                "display": "dropdown",
+                "label": "SharePoint data source",
+                "options": [
+                    {"label": SHAREPOINT_ONLINE_LABEL, "value": SHAREPOINT_ONLINE},
```

Comment:
nit: If we use them only one time I think we can inline `SHAREPOINT_ONLINE_LABEL` and `SHAREPOINT_SERVER_LABEL`. But feel free to keep it like that, I try to only use constants, if the strings appear more than once or the value is too cryptic to understand it directly (like numbers, regular expressions etc.).